src/Package/Impl/DataInspect/Commands/DeleteAllVariablesCommand.cs

```diff
@@ -8,13 +8,19 @@
 using Microsoft.VisualStudio.R.Package.Commands;
 using Microsoft.VisualStudio.R.Package.Shell;
 using Microsoft.VisualStudio.R.Packages.R;
+using Microsoft.VisualStudio.R.Package.Utilities;
 
 namespace Microsoft.VisualStudio.R.Package.DataInspect.Commands {
     internal sealed class DeleteAllVariablesCommand : SessionCommand {
         public DeleteAllVariablesCommand(IRSession session) :
             base(session, RGuidList.RCmdSetGuid, RPackageCommandId.icmdDeleteAllVariables) {
         }
 
+        protected override void SetStatus() {
+            var variableWindowPane = ToolWindowUtilities.FindWindowPane<VariableWindowPane>(0);
+            Enabled = (variableWindowPane?.IsGlobalREnvironment()) ?? true;
```

Comment:
I will update this. I think the button should be enabled only when it can be used.